diff --git a/networkd/network_server.ml b/networkd/network_server.ml
index c5b3f0b..4f1a9b4 100644
--- a/networkd/network_server.ml
+++ b/networkd/network_server.ml
@@ -387,7 +387,7 @@ module Bridge = struct
 
 	let determine_backend () =
 		let backend = String.strip String.isspace
-			(Unixext.string_of_file ("/etc/xcp/network.conf")) in
+			(Unixext.string_of_file ("/etc/xensource/network.conf")) in
 		match backend with
 		| "openvswitch" | "vswitch" -> kind := Openvswitch
 		| "bridge" -> kind := Bridge
