From 67f6df1e74819b6583dbdf948640196ca58650a9 Mon Sep 17 00:00:00 2001
From: Si Beaumont <simon.beaumont@citrix.com>
Date: Fri, 18 Dec 2015 10:33:44 +0000
Subject: [PATCH] (experimental) CA-190329: Add watch on attr Xenstore key

Signed-off-by: Si Beaumont <simon.beaumont@citrix.com>
---
 xc/xenops_server_xen.ml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xc/xenops_server_xen.ml b/xc/xenops_server_xen.ml
index fca5025..bf7839a 100644
--- a/xc/xenops_server_xen.ml
+++ b/xc/xenops_server_xen.ml
@@ -2400,6 +2400,7 @@ module Actions = struct
 
 	let interesting_paths_for_domain domid uuid =
 		let open Printf in [
+			sprintf "/local/domain/%d/attr" domid;
 			sprintf "/local/domain/%d/data/updated" domid;
 			sprintf "/local/domain/%d/data/ts" domid;
 			sprintf "/local/domain/%d/memory/target" domid;
-- 
1.9.1

